### DISCUSSION TITLE
Custom feedback types

### DISCUSSION THREAD

**Discussion by justoverclockl** [BetaHuhn/feedback-js#51] [Ideas] [answered]
First of all, how can i customize the feedback type?

 - General Feedback
 - I have an idea
 - Found a bug

and how can i send the form without backend endpoint?

**Comment by BetaHuhn** [comment]
Hey @justoverclockl,

> how can i customize the feedback type?

There is currently no way to customize the feedback types, they are hard coded. This should be a simple fix though, if you want you can create a PR for this!

> how can i send the form without backend endpoint?

What do you mean by this? Where do you want to send the form?

**Reply by justoverclockl** [reply]
oh, so i think i need to drop this package for now :(

**Reply by BetaHuhn** [reply]
I would be willing to add custom feedback types if this is something you need.

**Reply by justoverclockl** [reply]
yes this will be a cool solution :)

to make this great could u implement a backend part that allow us to use custom gmail account to send the form? i'm not expert but these could be perfect

**Reply by BetaHuhn** [reply]
There are already a lot of services available which do this. Most of them can be integrated with feedback-js, you only need to be able to send a POST request to them

**Reply by justoverclockl** [reply]
i will look at jamform, but the most important things is the customizable menu for feedback, i need to use this in various language and with various services. i will wait for your implementation :)

**Reply by BetaHuhn** [reply]
I will work on this later today, should be quick to implement.

**Reply by justoverclockl** [reply]
after your update how can i update your package?

**Reply by BetaHuhn** [reply]
It depends on how you are using it. If you are loading it via the CDN it should update automatically, via NPM run `npm install @betahuhn/feedback-js` again.

**Comment by BetaHuhn** [comment, marked as answer]
Added in [v2.0.0](https://github.com/BetaHuhn/feedback-js/releases/tag/v2.0.0) 🎉 

Use the `types` option like this:

```js
import Feedback from '@betahuhn/feedback-js'

const options = {
    endpoint: 'https://example.com/feedback',
    types: {
        general: {
            text: 'General Feedback',
            icon: '⚠️'
        },
        idea: {
            text: 'I have an idea',
            icon: '💡'
        },
        love: {
            text: 'Send love',
            icon: '💖'
        }
    }
}

new Feedback(options).renderButton()
```
